lib/absinthe/introspection/kind.ex

```diff
@@ -8,19 +8,20 @@ defmodule Absinthe.Introspection.Kind do
         __MODULE__
         |> Module.split()
         |> List.last()
-        |> Absinthe.Introspection.Kind.upcase()
+        |> Absinthe.Introspection.Kind.downcase()
```

Comment:
Given this is enumerated now, and not just a string, I wonder if this can be simplified (ie: not use all that module / string manipulation)

It seems like the goal is just to have a function `kind` that returns an atom... We could get rid of the macro altogether:

```elixir
defmodule Absinthe.Type.Object do
  @behaviour Absinthe.Introspection.Kind
  def kind, do: :object
end

defmodule Absinthe.Introspection.Kind do
  @type introspection_kind :: :enum | :object # ...
  @callback kind() :: introspection_kind()
end
```

Reply:
I think I'll do this separately...

Reply:
Actually while making this change, I found some further issues with our `__TypeKind`!

Further corrections here: https://github.com/absinthe-graphql/absinthe/pull/1019

+ 1 to being explicit!

Reply:
Thx, good catch!